extensions/agent/src/data/jobStepData.ts

```diff
@@ -157,4 +158,115 @@ export class JobStepData implements IAgentDialogData {
 		return result;
 	}
 
+	public static convertToAgentSubSystem(subSystemDisplayName: string): azdata.AgentSubSystem {
+		switch (subSystemDisplayName) {
+			case (JobStepDialog.TSQLScript): {
+				return azdata.AgentSubSystem.TransactSql;
+			}
+			case (JobStepDialog.Powershell): {
+				return azdata.AgentSubSystem.PowerShell;
+			}
+			case (JobStepDialog.CmdExec): {
+				return azdata.AgentSubSystem.CmdExec;
+			}
+			case (JobStepDialog.ReplicationDistributor): {
+				return azdata.AgentSubSystem.Distribution;
+			}
+			case (JobStepDialog.ReplicationMerge): {
+				return azdata.AgentSubSystem.Merge;
+			}
+			case (JobStepDialog.ReplicationQueueReader): {
+				return azdata.AgentSubSystem.QueueReader;
+			}
+			case (JobStepDialog.ReplicationSnapshot): {
+				return azdata.AgentSubSystem.Snapshot;
+			}
+			case (JobStepDialog.ReplicationTransactionLogReader): {
+				return azdata.AgentSubSystem.LogReader;
+			}
+			case (JobStepDialog.AnalysisServicesCommand): {
+				return azdata.AgentSubSystem.AnalysisCommands;
+			}
+			case (JobStepDialog.AnalysisServicesQuery): {
+				return azdata.AgentSubSystem.AnalysisQuery;
+			}
+			case (JobStepDialog.ServicesPackage): {
+				return azdata.AgentSubSystem.Ssis;
+			}
+			default:
+				return azdata.AgentSubSystem.TransactSql;
+		}
+	}
+
+	public static convertToSubSystemDisplayName(subSystem: azdata.AgentSubSystem): string {
```

Comment:
I'd like to see a unit test that goes through all the values of azdata.AgentSubSystem and calls convertToSubSystemDisplayName so we can verify all options have a mapping.